src/main/kotlin/com/learnspigot/bot/command/StartvoteCommand.kt

```diff
@@ -0,0 +1,123 @@
+    private val leaderboardManager: LeaderboardManager
+) {
+    fun startvoteCommand() {
+        guild.upsertCommand("startvote", "Start a vote for project or tutorial") {
+            restrict(guild = true)
+            bot.onCommand("startvote") {
+                if (guild.getMemberById(it.user.id)!!.roles.contains(guild.getRoleById(System.getenv("SUPPORT_ROLE_ID"))) && bot.getThreadChannelById(
+                        it.channel!!.id
+                    )?.parentChannel?.id != null && bot.getThreadChannelById(
+                        it.channel!!.id
+                    )!!.parentChannel.id == System.getenv("FOR_REVIEW_CHANNEL_ID")
+                ) {
+                    it.replyEmbed({
+                        title = "Success"
+                        description = "A vote has been started, results will be announced in 24 hours"
+                        color = EMBED_COLOR
+                    }).queue()
+                    val tutorialEmojis = mutableListOf("0️⃣", "1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣")
+                    val projectEmojis =
+                        mutableListOf("0️⃣", "1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣", "6️⃣", "7️⃣", "8️⃣", "9️⃣", "🔟")
+                    val type = bot.getThreadChannelById(it.channel!!.id)!!.appliedTags[0].name
+                    var message: Message? = null
+                    bot.getTextChannelById(System.getenv("VOTE_CHANNEL_ID"))!!
+                        .sendMessageEmbeds(voteEmbed(type, it, false)).queue { m ->
+                            message = m
+                            if (type == "Tutorial") tutorialEmojis.forEach { e ->
+                                m.addReaction(Emoji.fromUnicode(e)).queue()
+                            }
+                            else projectEmojis.forEach { e -> m.addReaction(Emoji.fromUnicode(e)).queue() }
+                        }
+                    Timer().schedule(86400000) {
+                        var sum = 0
+                        val usersReacted = mutableListOf<String>()
+                        val usersReaction = mutableListOf<Int>()
+                        if (type == "Tutorial") {
+                            for (emo in tutorialEmojis.indices.reversed()) {
+                                for (user in bot.getTextChannelById(System.getenv("VOTE_CHANNEL_ID"))!!
+                                    .retrieveMessageById(message!!.id).complete()
+                                    .retrieveReactionUsers(Emoji.fromUnicode(tutorialEmojis[emo]))) if (!usersReacted.contains(
+                                        user.id
+                                    ) && user.id != bot.selfUser.id
+                                ) {
+                                    usersReacted.add(user.id)
+                                    usersReaction.add(emo)
+                                }
+                            }
+                        } else {
+                            for (emo in projectEmojis.indices.reversed()) {
```

Comment:
Also this looks very duplicated, is there anyway the logic can be shared?

Reply:
not really, tutorial and project got different amount of rep to be added and it got different variables, and i reverse it just to have the highest score in highest priority